src/docs/spec-model.md

```diff
@@ -92,6 +92,25 @@ In addition, predicates from the following sources MAY also be encouraged:
 * any relation under [skos:mappingRelation](http://www.w3.org/2004/02/skos/core#mappingRelation) in the [Semantic Mapping Vocabulary](https://mapping-commons.github.io/semantic-mapping-vocabulary/).
 
 
+## Literal mappings
+
+<a id="literal-mappings"></a>
+
+The SSSOM model is primarily intended to represent mappings between semantic entities. However, it may also be used to represent mappings where at least one side is a literal string that does not have an identifier of its own. Any such mapping is henceforth called a _literal mapping_.
+
+To represent a mapping whose subject (resp. object) is a literal:
+
+* the `subject_type` (resp. `object_type`) slot is set to `rdfs literal`;
+* the `subject_label` (resp. `object_label`) slot is set to the literal itself;
+* the `subject_id` (resp. `object_id`) slot is left empty.
```

Comment:
Because this list is not using `MUST` and `SHOULD` phrases, it is left unclear if you can switch to literal mode and still have something in the id slot. Thinking of this a bit, I actually would, counterintuitively, favour this slightly, because it would give me a natural way to encode enum mappings (no need to be normative here). I also do not want to, right this moment, think it through if this is a bad idea or not (it might be).

subject_type: "rdfs literal"
subject_id: http://w3id.org/example/clinicaldatamodel/gender
subject_label: "FEMALE"
object_id: `PATO:0000383`

Again, I am not saying this is a good idea, but maybe its worth keeping the door open?

The alternative would be to have a seperate slot, `subject_label_context` or some such, in a future version.

In any case, what is actionable:

Either we should clearly say that it `MUST` be empty, or clearly say `SHOULD` be empty.

Reply:
> still have something in the id slot. Thinking of this a bit, I actually would, counterintuitively, favour this slightly

So do I. For at least two reasons:

* Symmetry: For a non-literal mapping (where what matters is the `subject/object_id`), it is of course still perfectly valid to have a `subject/object_label`, so I feel like it should also be allowed, for a literal mapping (where what matters is the `subject/object_label`), to have something in `subject/object_id`.
* Somehow, even a literal could have _some kind_ of identifier, which may not be a “proper” semantic identifier (for example, it may be only meaningful for the person who compiled the list of literals) but that you may still want to record.

For example, if I map a list of scRNAseq cluster names to CL cell types, I _may_ want to do something like this:

```
#subject_type: rdfs literal
#object_source: obo:CL
subject_id   subject_label   predicate_id      object_id
cluster:1    astrocyte       skos:exactMatch   CL:1234567
cluster:2    oligodendrocyte skos:exactMatch   CL:2345678
...
```

These are “literal mappings” because the clusters are not really entities in any database or ontology, they are just entries in a local list that is only meaningful to me (at least until the dataset is published), and what is important for the matching here is really the name I gave to those clusters.

Allowing this kind of “pseudo-IDs” would also make it easier to convert those mappings to proper entity-to-entity mappings if and once the clusters get somehow published later and get assigned proper permanent IDs.

> Either we should clearly say that it MUST be empty, or clearly say SHOULD be empty.

I’d rather say “MAY be left empty”. No reason to strongly recommend that it should be left empty. The point is that the `subject/object_id` slot is basically _irrelevant_ for literal mappings, so do whatever you want with it (fill it or leave it empty, it doesn’t matter).

Reply:
About “encoding enum mapping”:

> ```
> subject_type: "rdfs literal"
> subject_id: http://w3id.org/example/clinicaldatamodel/gender
> subject_label: "FEMALE"
> object_id: PATO:0000383
> ```

You mean using `subject_id` to identify the enum whose values you are mapping? Wouldn’t `subject_source` be more appropriate for that?

Reply:
Yes, probably!